status-im/status-react
# 🤝  [19834] Add dapp connection flow with a basic design

### Summary
This is an ongoing effort to implement wallet connect. This PR connects the existing methods to the universal qr code scanner. In case a valid Wallet Connect QR is scanned, the user gets an option to accept the connection.

In case the QR code is expired or the wallet connect version is not 2, the user gets a relevant error toast.

### Review notes

- Hard coded the first account
- The data item component doesn't support networks or accounts yet
- The quo category component cannot show a data-item yet
- The above two points mean that networks or accounts cannot be changed
- I didn't add the validation to ensure we support required namespaces
- Connected accept and decline button

#### Platforms


- Android
- iOS

#### Areas that maybe impacted
- Global QR scanner

### Steps to test
- Goto https://lab.web3modal.com/library/ethers-siwe/
- Select the mainnet in networks (this is important)
- Click Connect Wallet, a QR code will pop up
- Use the universal scanner to scan that QR code, you'll see a screen to accept or reject the session

status: ready
